packages/eslint-plugin-query/package.json

```diff
@@ -0,0 +1,45 @@
+{
+  "name": "@tanstack/eslint-plugin-query",
+  "version": "1.0.0",
+  "description": "ESLint plugin for React Query",
+  "author": "Eliya Cohen",
+  "license": "MIT",
+  "repository": "tanstack/query",
+  "homepage": "https://tanstack.com/query",
+  "funding": {
+    "type": "github",
+    "url": "https://github.com/sponsors/tannerlinsley"
+  },
+  "main": "build/lib/index.js",
+  "scripts": {
+    "typecheck": "tsc",
+    "build": "tsup --minify",
```

Comment:
for the build step to work and this to be published, it has to be included in `pnpm run build`, which currently executes this:

https://github.com/TanStack/query/blob/17a362b141b042f0fd9ff3ce849fd8858446fb30/package.json#L15

It's fine if we want to build this with tsup rather than rollup, though it being a dev dependency for consumers it doesn't really matter. I don't know the rollup config very well but if we could include it there, maybe that would be easier?

@DamianOsipiuk knows better how this works, and also knows if we have to add `module` and `exports` to `package.json` to make esm work?

Reply:
eslint doesn't work with esm. It has to be bundled as cjs.

Reply:
Yeah, it does not bring to much benefit to build `esm` for eslint right now, maybe in the future.
Nevertheless it would be neat to use the same build tools for all of the packages.
But i guess we could do that later, when we do some refactoring of current rollup config.

Reply:
> Nevertheless it would be neat to use the same build tools for all of the packages.

okay so can we switch to `rollup` just without ESM ?